ambry-mysql/src/main/java/com/github/ambry/accountstats/PartitionClassReportsDao.java

```diff
@@ -0,0 +1,294 @@
+    Set<Short> partitionIds = new HashSet<>();
+    try {
+      long startTimeMs = System.currentTimeMillis();
+      PreparedStatement queryStatement = dataAccessor.getPreparedStatement(queryPartitionIdsSql, false);
+      queryStatement.setString(1, clusterName);
+      try (ResultSet rs = queryStatement.executeQuery()) {
+        while (rs.next()) {
+          partitionIds.add((short) rs.getInt(1));
+        }
+      }
+      dataAccessor.onSuccess(Read, System.currentTimeMillis() - startTimeMs);
+      return partitionIds;
+    } catch (SQLException e) {
+      dataAccessor.onException(e, Read);
+      logger.error("Failed to execute query {}, with parameter {}", queryPartitionIdsSql, clusterName, e);
+      throw e;
+    }
+  }
+
+  /**
+   * Insert partition id with it's partition class id of given {@code clusterName}. If the partition id already
+   * exist for given {@code clusterName}, then this will be a no-op.
+   * @param clusterName The clusterName
+   * @param partitionId The id of the partition
+   * @param partitionClassId The id of the partition class name
+   * @throws SQLException
+   */
+  void insertPartitionId(String clusterName, short partitionId, short partitionClassId) throws SQLException {
+    try {
+      long startTimeMs = System.currentTimeMillis();
+      PreparedStatement insertStatement = dataAccessor.getPreparedStatement(insertPartitionIdSql, true);
+      insertStatement.setString(1, clusterName);
+      insertStatement.setInt(2, partitionId);
+      insertStatement.setShort(3, partitionClassId);
+      insertStatement.executeUpdate();
+      dataAccessor.onSuccess(Write, System.currentTimeMillis() - startTimeMs);
+    } catch (SQLIntegrityConstraintViolationException e) {
+      // If somehow the unique pair(clusterName, partitionId) is inserted, consider this as a success
+      // ignore it
+    } catch (SQLException e) {
+      dataAccessor.onException(e, Write);
+      logger.error("Failed to execute of {}, with parameter {} {} {}", insertPartitionIdSql, clusterName, partitionId,
+          partitionClassId, e);
+      throw e;
+    }
+  }
+
+  /**
+   * Query container storage usage for given {@code clusterName}. This usage is aggregated under each partition class name.
+   * The results will be applied to the given {@link PartitionClassContainerUsageFunction}.
+   * @param clusterName The clusterName.
+   * @param func The callback to apply query results
+   * @throws SQLException
+   */
+  void queryAggregatedPartitionClassReport(String clusterName, PartitionClassContainerUsageFunction func)
+      throws SQLException {
+    try {
+      long startTimeMs = System.currentTimeMillis();
+      PreparedStatement queryStatement = dataAccessor.getPreparedStatement(queryAggregatedSql, false);
+      queryStatement.setString(1, clusterName);
+      try (ResultSet rs = queryStatement.executeQuery()) {
+        while (rs.next()) {
+          String className = rs.getString(1);
```

Comment:
Minor: please call this partitionClassName since className means something else in java.